### PR TITLE
Add "null" counter to facets (fixes #99)

### LINKED ISSUE #99: Add None to Facets for when the Data is not Tagged
┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/AZUL-66)
┆Project Name: azul
┆Issue Number: AZUL-66

### PR DESCRIPTION
**Pull request by Ben-Wu**
(fixes #99)

This pull request adds a counter for facets that do not have a value in the ES aggregation.  The response will have in `termFacets` something like this (using `null` value as the term):

```js
{
  "biologicalSex":{
    "terms":[
      {
        "term":"female",
        "count":6220
      },
      {
        "term":"unknown",
        "count":4
      },
      {
        "term":null,
        "count":10
      }
    ]
  }
}
```

Another change is to allow querying for missing values through the api.  This is done with a filter in the query like:
```
{"file":{"fieldA":{"is": None}}}
```
Let me know if you have ideas on how to make this query better.

@hannes-ucsc Also as per our previous discussion let me know if this should be separated into multiple PRs as the changes aren't really dependent on each other.

### CODE REVIEW

File: src/azul/service/responseobjects/elastic_request_builder.py [hunk 1]
```diff
@@ -78,11 +78,17 @@ def create_query(filters):
         translated (es_keys) keys
         :return: Returns Query object with appropriate filters
         """
+        filter_list = []
+        # If filter is {"is": None}, search for values where field does not exist
+        for facet, values in filters.items():
+            if values.get('is', {}) is None:
+                f = Q('bool', must_not=Q('exists', field=f'{facet}.keyword'))
+            else:
+                f = Q('terms', **{f'{facet.replace(".", "__")}__keyword': values.get('is')})
```
[the remark this answers is not in the document]
Comment (Ben-Wu) [reply src/azul/service/responseobjects/elastic_request_builder.py]: Oh thanks for catching that.  I think I got mixed up by `if values.get('is', {}) is None:` and thought  the second `.get` would always return a value.
Looks like I also missed a test case where `is` is missing from the filter.

**Comment by Ben-Wu** [comment]
@hannes-ucsc About that `values.get('is', {})`, it looks like if you do that and pass an empty dict in the filter an error is raised.

What is the expected behaviour of passing in a filter like {"file": {"organismAge": {}}}?
Should it crash? Or should it return no results or return all the values?

I think crashing makes sense if it's an invalid input but I think it's strange to do it this way.

**Comment by hannes-ucsc** [comment]
You've retained the existing behavior for `is` being absent and that's what we should concerned with here. I will create a separate ticket.

**Comment by hannes-ucsc** [comment]
https://github.com/DataBiosphere/azul/issues/280

**Pull request merged**